cylc/flow/scripts/lint.py

```diff
@@ -1417,6 +1434,14 @@ def main(parser: COP, options: 'Values', target=None) -> None:
         max_line_len=mergedopts[MAX_LINE_LENGTH]
     )
 
+    if options.rule:
+        checks = {k: v for k, v in checks.items() if k in options.rule}
```

Comment:
I think this can be simplified to:

```python
checks = {rule: checks[rule] for rule in options.rule}
```

Which has the advantage of raising an error if a rule does not exist.

```suggestion
        try:
            checks = {rule: checks[rule] for rule in options.rule}
        except KeyError as exc:
            LOG.error(f'No such rule {exc.args[0]}')
            sys.exit(1)
```

However, it looks like `--rule` can only accept one rule at present?

I feel we should implement this more like `--select` (ruff/flake8), to allow the selection of rules OR rule sets?